frontend/src/metabase/static-viz/lib/axes.js

```diff
@@ -26,24 +22,37 @@ export const getYTickWidth = (data, accessors, settings, fontSize) => {
     .reduce((a, b) => Math.max(a, b), 0);
 };
 
-export const getXTickLabelProps = (layout, isVertical, getColor) => ({
-  fontSize: layout.font.size,
-  fontFamily: layout.font.family,
-  fill: getColor("text-medium"),
+/**
+ *
+ * @param {import("../components/XYChart/types").ChartStyle} chartStyle
+ * @param {boolean} isVertical
+ */
+export const getXTickLabelProps = (chartStyle, isVertical) => ({
```

Comment:
`chartStyle` is defined here:
https://github.com/metabase/metabase/blob/0fd17b376cb02a0337e5ae2c5bcd6cd65f5b1a8e/frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx#L74-L95

which is almost the same style defined in `LineAreaBarChart`:
https://github.com/metabase/metabase/blob/0fd17b376cb02a0337e5ae2c5bcd6cd65f5b1a8e/frontend/src/metabase/static-viz/components/LineAreaBarChart/LineAreaBarChart.tsx#L24-L50

except some unused properties are removed e.g. `goalColor`, `legend`